File: Medium/1641-Count Sorted Vowel Strings/Dynamic Programming.rb
# @param {Integer} n
# @return {Integer}
def count_vowel_strings(n)
  dp = [1] * 5

  (1...n).each do |_|
    dp[1] += dp[0]
    dp[2] += dp[1]
    dp[3] += dp[2]
    dp[4] += dp[3]
  end

  dp.sum
end

File: Medium/1641-Count Sorted Vowel Strings/Dynamic Programming.rs
impl Solution {
    pub fn count_vowel_strings(n: i32) -> i32 {
        let mut dp = [1; 5];

        for _ in 1..n {
            dp[1] += dp[0];
            dp[2] += dp[1];
            dp[3] += dp[2];
            dp[4] += dp[3];
        }

        dp.iter().sum()
    }
}
